plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,321 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package slurm
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Slurm struct {
+	URL      string `toml:"url"`
+	Username string `toml:"username"`
+	Token    string `toml:"token"`
+
+	IgnoreDiag         bool `toml:"ignoreDiag"`
+	IgnoreJobs         bool `toml:"ignoreJobs"`
+	IgnoreNodes        bool `toml:"ignoreNodes"`
+	IgnorePartitions   bool `toml:"ignorePartitions"`
+	IgnoreReservations bool `toml:"ignoreReservations"`
+
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+	tls.ClientConfig
+
+	client  *goslurm.APIClient
+	baseURL *url.URL
+}
+
+func (s *Slurm) createHTTPClient(host string) (*goslurm.APIClient, error) {
+	configuration := goslurm.NewConfiguration()
+	configuration.Host = host
+	configuration.Scheme = "http"
+	configuration.UserAgent = "Telegraf Metrics Agent"
+	configuration.HTTPClient = &http.Client{
+		Timeout: time.Duration(s.ResponseTimeout),
+	}
+
+	return goslurm.NewAPIClient(configuration), nil
+}
+
+func (s *Slurm) createHTTPSClient(host string) (*goslurm.APIClient, error) {
+	tlsCfg, err := s.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	configuration := goslurm.NewConfiguration()
+	configuration.Host = host
+	configuration.Scheme = "https"
+	configuration.UserAgent = "Telegraf Metrics Agent"
+	configuration.HTTPClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+		},
+		Timeout: time.Duration(s.ResponseTimeout),
+	}
+
+	return goslurm.NewAPIClient(configuration), nil
+}
+
+func (*Slurm) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Slurm) Init() error {
+	if s.ResponseTimeout < config.Duration(time.Second) {
+		s.ResponseTimeout = config.Duration(time.Second * 5)
+	}
+
+	if s.URL == "" {
+		return fmt.Errorf("empty URL provided")
+	}
+
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		return err
+	}
+
+	if u.Hostname() == "" {
+		return fmt.Errorf("invalid hostname %q", u.Hostname())
+	}
+
```

Comment:
Why do we need to check for an invalid hostname if we have already successfully parsed the URL?

Reply:
I prepared an example [here](https://play.golang.com/p/K_1cFDaJkTx) showing that one might get an invalid hostname even if `url.Parse()` doesn't return an error. That's the corner case I was trying to guard against with this check...

Reply:
ah thanks for that

Reply:
Since the hostname will always be empty in the error message, maybe add some better context?
```suggestion
	if u.Hostname() == "" {
		return fmt.Errorf("empty hostname for url %q", s.URL)
	}

```